cfme/tests/openstack/cloud/test_keypairs.py

```diff
@@ -0,0 +1,27 @@
+"""tests for Openstack cloud Keypairs"""
+
+import fauxfactory
+import pytest
+
+from cfme.cloud.provider.openstack import OpenStackProvider
+
+
+pytestmark = [
+    pytest.mark.usefixtures("setup_provider"),
+    pytest.mark.provider([OpenStackProvider], scope='function')
+]
+
+
+@pytest.fixture(scope='function')
+def keypair(appliance, provider):
+    keypair = appliance.collections.cloud_keypairs.create(name=fauxfactory.gen_alphanumeric(),
+                                                          provider=provider)
+    yield keypair
+    if keypair.exists:
+        keypair.delete(wait=False)
+
+
+@pytest.mark.ignore_stream('5.9')
+def test_download_private_key(keypair):
+    assert keypair.exists
+    keypair.download_private_key()
```

Comment:
Required:
Need to validate downloaded or not? Flash message assertion needed in the test as in model; we just checked `no_error`

Reply:
@jadhaj look at this. It will give you an idea:
https://github.com/ManageIQ/integration_tests/blob/master/cfme/tests/intelligence/test_download_report.py#L49

Reply:
I am not sure I got the point. May be you mean that i need to implement download method like this:
https://github.com/ManageIQ/integration_tests/blob/master/cfme/intelligence/reports/reports.py#L162

or might be using need_firefox will prevent popup Save system window to appear and it will be displayed as web view

Also wait_for(lambda: any([file.endswith(extension) for file in os.listdir(
        browser.firefox_profile_tmpdir)]), num_sec=60.0)

assumes save button was clicked which i don't see where it is clicked actually

please shed more light

Reply:
@jadhaj I was talking about `firefox_profile_tmpdir` access. When you will download the file. You will able to locate it under it and easily assert. This assertion will complete your test.